Source/Core/Core/PowerPC/PPCCache.h

```diff
@@ -12,37 +12,64 @@ class PointerWrap;
 
 namespace PowerPC
 {
-constexpr u32 ICACHE_SETS = 128;
-constexpr u32 ICACHE_WAYS = 8;
+constexpr u32 CACHE_SETS = 128;
+constexpr u32 CACHE_WAYS = 8;
 // size of an instruction cache block in words
-constexpr u32 ICACHE_BLOCK_SIZE = 8;
+constexpr u32 CACHE_BLOCK_SIZE = 8;
 
-constexpr u32 ICACHE_EXRAM_BIT = 0x10000000;
-constexpr u32 ICACHE_VMEM_BIT = 0x20000000;
+constexpr u32 CACHE_EXRAM_BIT = 0x10000000;
+constexpr u32 CACHE_VMEM_BIT = 0x20000000;
 
-struct InstructionCache
+struct Cache
 {
-  std::array<std::array<std::array<u32, ICACHE_BLOCK_SIZE>, ICACHE_WAYS>, ICACHE_SETS> data{};
-  std::array<std::array<u32, ICACHE_WAYS>, ICACHE_SETS> tags{};
-  std::array<u32, ICACHE_SETS> plru{};
-  std::array<u32, ICACHE_SETS> valid{};
+  std::array<std::array<std::array<u32, CACHE_BLOCK_SIZE>, CACHE_WAYS>, CACHE_SETS> data{};
+
+  // Stores the 32-byte aligned address of the start of each cache block. This consists of the cache
+  // set and tag. Real hardware only needs to store the tag, but also including the set simplifies
+  // debugging and getting the actual address in the cache, without changing behavior (as the set
+  // portion of the address is by definition the same for all addresses in a set).
+  std::array<std::array<u32, CACHE_WAYS>, CACHE_SETS> addrs{};
+
+  std::array<u32, CACHE_SETS> plru{};
+  std::array<u32, CACHE_SETS> valid{};
+  std::array<u32, CACHE_SETS> modified{};
```

Comment:
Since there are 8 ways, these can be an `std::array<u8, CACHE_SETS>`. This was a change I made in #10826, but I'm not sure why I didn't mention it earlier. It'd be nice to make this change now to avoid bumping savestate versions in other PRs.

I'm not sure why it was originally written as `u32`.